### ISSUE TITLE
Rename this repo to "did-jose"

### ISSUE THREAD

**Issue by oed** [decentralized-identity/did-jwt#170]
This package now contains many things related to the JOSE spec. We should modularize it to a few different (to be determined) packages.

Suggestion:
* `@did-jose/jws`
* `@did-jose/jwe`
* `@did-jose/jwt`
* `@did-jose/signing-*`
* `@did-jose/encryption-*`

**Comment by peacekeeper** [comment]
Hey, since this is a work item of the I&D WG, if you rename it then maybe you could also update the entry in the [list of Work Items](https://github.com/decentralized-identity/identifiers-discovery/#work-items), and while doing that also check the work item leads that are listed there?

Perhaps we will also ask you at some point to give a general update on the work item progress during one of the weekly I&D WG calls.

**Comment by TomCJones** [comment]
I am unclear where this label is applied. Jose has a header.  Some of what is proposed is already in that header. Perhaps all of it should be in the Jose header. Otherwise difference levels of the code would be involved. That seems wrong.

**Comment by oed** [comment]
@TomCJones Are you replying to the wrong issue? :)

**Comment by awoie** [comment]
I agree that renaming/repackaging makes sense but I don't have a strong stance here.

**Comment by oed** [comment]
@mirceanis Would love to hear your thoughts here!

**Comment by mirceanis** [comment]
Modularising this repo sounds like a good idea.
I'm not yet sure about the actual module layout.
I think the goals should be to reduce lib size (and transitive dependencies) for users that only use a subset of modules, and to allow extension of the supported algorithms using user-provided modules.

**Comment by oed** [comment]
> I think the goals should be to reduce lib size (and transitive dependencies) for users that only use a subset of modules, and to allow extension of the supported algorithms using user-provided modules.

Yep agreed! That was what I was aiming for above, e.g. with `@did-jose/signing-*` representing a specific alg.

**Comment by oed** [comment]
This is not stale.
